### PR TITLE
feat(shared): zod-typed routes — auth/pair + inbox/messages

### PR DESCRIPTION
**Pull request by Dexploarer**
## Summary

First sweep beyond \`apps-routes.ts\`. Two small but useful surfaces: device pairing + inbox replies.

| Route | Schema |
|---|---|
| \`POST /api/auth/pair\` | \`PostAuthPairRequestSchema\` (request) + \`PostAuthPairResponseSchema\` (response) |
| \`POST /api/inbox/messages\` | \`PostInboxMessageRequestSchema\` (request side only — see below) |

## Pattern recap

Same template as the apps-routes sweeps (#7557 / #7559 / #7561 / #7565):

- Schema in \`packages/shared/src/contracts/<area>-routes.ts\`, \`.strict()\` plus \`.transform(trim)\` where the legacy hand-rolled validation did normalisation.
- Tests next to the schema cover: required-field rejection, trim/case semantics, strict-extra rejection, type-mismatch rejection.
- Server wires the schema via \`safeParse\`; on failure returns a 400 with the failing path + zod issue message. On success, destructures \`parsed.data\` and proceeds.

## Auth specifics

\`normalizePairingCode\` (whitespace strip + uppercase) still runs on the schema-validated \`code\` so existing clients that send padded codes continue to work. The schema only enforces non-empty + string at the wire.

## Inbox specifics

The schema's \`.transform()\` does the lowercase + trim work the handler used to do by hand (\`source.trim().toLowerCase()\`, \`text.trim()\`, etc.). Downstream checks (\`runtimeHasSendHandler\`, \`getRoom\`) are unchanged — the schema only handles wire-shape validation.

Response shape (\`{ ok: true, message?: InboxMessage }\`) intentionally **not** modelled here. \`InboxMessage\` is a large runtime-internal type and the inbox surface is mid-refactor; same deferral pattern PR #7561 used for routes that delegate to deeper TS interfaces.

## Out of scope

\`POST /api/onboarding\` was the third candidate but its body covers ~20 optional fields with normalisation helpers and a legacy-field rejection list. Deferred to a dedicated focused PR.

## Test plan

- [x] \`bun run test\` in \`@elizaos/shared\` — 14 files, 250 tests pass (18 new)
- [x] \`bun turbo run build --filter=@elizaos/shared\` — clean
- [x] \`bun run typecheck\` in \`@elizaos/agent\` — no new errors
- [x] Biome lint clean
- [ ] Reviewer: spot-check that the legacy clients sending padded pairing codes still pass through \`normalizePairingCode\` after schema validation

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<details><summary><h3>Greptile Summary</h3></summary>

This PR extends the Zod-typed routes sweep to two new surfaces: `POST /api/auth/pair` and `POST /api/inbox/messages`. Each gets a contract schema in `packages/shared/src/contracts/`, wired into the existing route handlers via `safeParse`, with matching test files.

- **Auth pair route** (`auth-routes.ts`/`auth-routes.test.ts`): `PostAuthPairRequestSchema` enforces a non-empty string `code`; `normalizePairingCode` still runs post-validation for legacy-client padding. Response typed via `PostAuthPairResponse`.
- **Inbox messages route** (`inbox-routes.ts`/`inbox-routes.test.ts`): `PostInboxMessageRequestSchema` replaces the hand-rolled trim/lowercase guards with a `.strict().transform()` chain; whitespace-only `replyToMessageId` is correctly absorbed by the transform rather than rejected at the wire.
</details>

<h3>Confidence Score: 5/5</h3>

Safe to merge — the changes are additive schema validations with no behavioral regressions.

Both route handlers now reject malformed inputs at the wire boundary with clear 400 messages, replacing ad-hoc hand-rolled checks. The whitespace-only replyToMessageId edge case flagged in a previous review round is handled correctly in the transform, and a dedicated test covers it. The normalizePairingCode post-validation step for the auth route is preserved as designed.

No files require special attention.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/shared/src/contracts/auth-routes.ts | New file: clean Zod schema for auth/pair request and response; .strict() prevents extra fields. |
| packages/shared/src/contracts/inbox-routes.ts | New file: inbox message request schema with regex-based non-whitespace validation and transform; whitespace-only replyToMessageId correctly dropped rather than rejected. |
| packages/agent/src/api/auth-routes.ts | Handler wired to PostAuthPairRequestSchema via safeParse; rawBody === null guard replaces !body; normalizePairingCode still applied post-validation as intended. |
| packages/agent/src/api/inbox-routes.ts | Handler wired to PostInboxMessageRequestSchema; manual trim/lowercase replaced by schema transform; rawBody === null guard is correct. |
| packages/shared/src/contracts/auth-routes.test.ts | New file: covers required-field, empty, non-string, strict-extra rejection for both request and response schemas. |
| packages/shared/src/contracts/inbox-routes.test.ts | New file: 15 test cases including whitespace-only for all required fields plus both whitespace-only and empty replyToMessageId absence cases. |
| packages/shared/src/contracts/index.ts | Adds re-exports for auth-routes.js and inbox-routes.js in alphabetical order. |

</details>

</details>

<details><summary><h3>Sequence Diagram</h3></summary>

```mermaid
sequenceDiagram
    participant C as Client
    participant H as Route Handler
    participant S as Zod Schema
    participant L as Business Logic

    C->>H: "POST /api/auth/pair"
    H->>H: readJsonBody (null check)
    H->>S: safeParse(rawBody)
    alt invalid shape
        S-->>H: success false
        H-->>C: 400 Bad Request
    else valid
        S-->>H: parsed.data.code
        H->>L: normalizePairingCode + timingSafeEqual
        L-->>H: match result
        H-->>C: 200 token
    end

    C->>H: "POST /api/inbox/messages"
    H->>H: readJsonBody (null check)
    H->>S: safeParse + trim/lowercase transform
    alt invalid shape
        S-->>H: success false
        H-->>C: 400 Bad Request
    else valid
        S-->>H: roomId, source, text, replyToMessageId
        H->>L: runtimeHasSendHandler + getRoom + sendMessageToTarget
        L-->>H: result
        H-->>C: 200 ok
    end
```
</details>

<sub>Reviews (2): Last reviewed commit: ["fix(shared): treat whitespace-only reply..."](https://github.com/elizaos/eliza/commit/10f7c565235c8a046337a81015cb666883ebff1d) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=31531406)</sub>

### CODE REVIEW

File: packages/shared/src/contracts/inbox-routes.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  }))
+  .pipe(
+    z
+      .object({
+        roomId: z.string().min(1, "roomId is required"),
+        source: z.string().min(1, "source is required"),
+        text: z.string().min(1, "text is required"),
+        replyToMessageId: z.string().min(1).optional(),
+      })
+      .strict(),
```
[the remark this answers is not in the document]
Comment (Dexploarer) [reply packages/shared/src/contracts/inbox-routes.ts]: Fixed in 10f7c5652 — split the validation: required fields stay with `regex(/\S/)`, optional `replyToMessageId` accepts any string and the transform treats whitespace-only the same as absent (omitted from the parsed output, not 400'd). Test added to lock the new behavior.

File: packages/shared/src/contracts/inbox-routes.test.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    expect(parsed.replyToMessageId).toBe("msg-2");
+  });
+
+  it("trims roomId, text, and replyToMessageId", () => {
+    const parsed = PostInboxMessageRequestSchema.parse({
+      roomId: "  room-1  ",
+      source: "TELEGRAM",
+      text: "  hello  ",
+      replyToMessageId: "  msg-2  ",
+    });
```
[the remark this answers is not in the document]
Comment (Dexploarer) [reply packages/shared/src/contracts/inbox-routes.test.ts:35]: Added in 10f7c5652 (now reframed as "treats whitespace-only replyToMessageId as absent" since the schema absorbs it instead of rejecting). Plus parallel whitespace cases for roomId and source for symmetry with the existing text test.
